contacts.mjs

```
@@ -0,0 +1,392 @@
+  const cjkLines = cjk.split('\r\n');
+  // 5 + 23*3 = 74: the 24th あ would hit 77 octets, so the line closes early
+  assert(Buffer.byteLength(cjkLines[0], 'utf-8') === 74, `CJK fold closes at 74 octets (never splits あ), got ${Buffer.byteLength(cjkLines[0], 'utf-8')}`);
+  assert(cjkLines.every(l => Buffer.byteLength(l, 'utf-8') <= 75), 'every folded CJK line <= 75 octets');
+  assert(cjkLines.map((l, i) => (i ? l.slice(1) : l)).join('') === 'NOTE:' + 'あ'.repeat(40), 'CJK content survives folding intact');
+
+  // UID — stable, deterministic slugs with hash fallback for empty slugs
+  assert(slug('Jörg Müller') === 'j-rg-m-ller', 'slug collapses non-alphanumerics');
+  assert(slug('--Acme  Inc.--') === 'acme-inc', 'slug trims dashes');
+  assert(uidPart('Jane Doe') === 'jane-doe', 'uidPart keeps the pretty slug for ASCII');
+  assert(/^[0-9a-f]{8}$/.test(uidPart('山田 太郎')), 'empty slug (CJK) falls back to 8-hex sha1');
+  assert(uidPart('山田 太郎') === uidPart('山田 太郎'), 'hash fallback deterministic');
```

Biome (2.5.3)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)